src/LandIce/evaluators/LandIce_IceOverburden_Def.hpp

```diff
@@ -79,25 +77,11 @@ template<typename EvalT, typename Traits>
 void IceOverburden<EvalT, Traits>::
 evaluateFieldsSide (typename Traits::EvalData workset)
```

Comment:
Similar idea in EffectivePressure, HydraulicPotential, BasalGravitationalWaterPotential, HydrologyMeltingRate, HydrologyResidualCavitiesEqn, HydrologyResidualMassEqn, HydrologyResidualTillStorageEqn, HydrologySurfaceWaterInput, HydrologyWaterDischarge, ScatterSideEqnResidualBase, FieldFrobeniusNorm . Some of them would need to be functions instead of kernels but the idea is the same. Just make sure that the cell implementation is doing the same thing as the side implementation.

Reply:
This is a good point and I think this is what is done with some of the other evaluators. I'll go through and combine the functions using `side_or_cell_idx` where possible and push an update to the PR.

Reply:
Jerry, will you take a look at the changes to IceOverburden and EffectivePressure that I just pushed? Do you prefer one approach or the other?

The difference is that in EffectivePressure, there is only one call to Kokkos::parallel_for and it takes a worksetSize that is set depending on if we are looking at the cell or sideset case.

Reply:
Imho, the version in IceOverburden is clearer.

Reply:
I think EffectivePressure looks cleaner. It's better to just have one parallel_for. Also I think it'd be better to use `cell` instead of `side_or_cell_idx` as I mentioned above.

Reply:
IceOverburden looks fine as is but if you do the same in EffectivePressure, I think you'd be left with 4 branches instead of 3.

Reply:
If you collapse the two kernels in EffectivePressure into one by moving `Surrogate` inside the kernel then it'd be the same as BasalFriction and IceOverburden. One downside to having multiple parallel_for is the build time since everything is inline but I don't think that's a huge concern right now.

Reply:
That seems to be the tradeoff. I find the IceOverburden way clearer as well but then it ends up with EffectivePressure needing to nest more if-elses. I think there are other evaluators in this batch that might have a similar situation to EffectivePressure so I am leaning towards the EffectivePressure approach if there are no objections.

Reply:
Ah, you're right, I can combine them into a single kernel and then at that point I think the IceOverburden way would be clearest. Should I go ahead with that approach?

Reply:
Yeah that's fine. I agree, it is clearer. I reserve the right to collapse them in the future if the build time get really bad. :)

Reply:
Ah, I forgot parallel_for was declared inline (why, I don't know.).

Reply:
Well, it's a moot point now anyway. As I was updating the other evaluators I came around to the `EffectivePressure` style so I'm going with that after all.

Reply:
Alright, I've pushed a new commit that simplifies a few of the other evaluators that have identical implementations for sides and cells. Let me know if it looks okay.